packages/dinero.js/src/api/__tests__/hasSubUnits.test.ts

```diff
@@ -13,84 +13,165 @@ import { hasSubUnits } from '..';
     });
   });
   describe('bigint', () => {
     const dinero = createBigintDinero;
     const bigintUSD = castToBigintCurrency(USD);
+    const bigintMGA = castToBigintCurrency(MGA);
+    const bigintGBP = { code: 'GBP', base: [20n, 12n], exponent: 1n };
 
-    it('returns false when there are no sub-units', () => {
-      const d = dinero({ amount: 1100n, currency: bigintUSD });
+    describe('decimal currencies', () => {
+      it('returns false when there are no sub-units', () => {
+        const d = dinero({ amount: 1100n, currency: bigintUSD });
 
-      expect(hasSubUnits(d)).toBe(false);
-    });
-    it('returns true when there are sub-units based on a custom scale', () => {
-      const d = dinero({ amount: 1100n, currency: bigintUSD, scale: 3n });
+        expect(hasSubUnits(d)).toBe(false);
+      });
+      it('returns true when there are sub-units based on a custom scale', () => {
+        const d = dinero({ amount: 1100n, currency: bigintUSD, scale: 3n });
 
-      expect(hasSubUnits(d)).toBe(true);
-    });
-    it('returns true when there are sub-units', () => {
-      const d = dinero({ amount: 1150n, currency: bigintUSD });
+        expect(hasSubUnits(d)).toBe(true);
+      });
+      it('returns true when there are sub-units', () => {
+        const d = dinero({ amount: 1150n, currency: bigintUSD });
+
+        expect(hasSubUnits(d)).toBe(true);
+      });
+      it('returns false when there are no sub-units based on a custom scale', () => {
+        const d = dinero({ amount: 1150n, currency: bigintUSD, scale: 1n });
 
-      expect(hasSubUnits(d)).toBe(true);
+        expect(hasSubUnits(d)).toBe(false);
+      });
     });
-    it('returns false when there are no sub-units based on a custom scale', () => {
-      const d = dinero({ amount: 1150n, currency: bigintUSD, scale: 1n });
+    describe('non-decimal currencies', () => {
+      it('returns false when there are no sub-units', () => {
+        const d = dinero({ amount: 10n, currency: bigintMGA });
+
+        expect(hasSubUnits(d)).toBe(false);
+      });
+      it('returns true when there are sub-units', () => {
+        const d = dinero({ amount: 11n, currency: bigintMGA });
+
+        expect(hasSubUnits(d)).toBe(true);
+      });
+      it('returns false when there are no sub-units based on a multi-base', () => {
+        const d = dinero({ amount: 240n, currency: bigintGBP });
+
+        expect(hasSubUnits(d)).toBe(false);
+      });
+      it('returns true when there are sub-units based on a multi-base', () => {
+        const d = dinero({ amount: 267n, currency: bigintGBP });
 
-      expect(hasSubUnits(d)).toBe(false);
+        expect(hasSubUnits(d)).toBe(true);
+      });
     });
   });
   describe('Big.js', () => {
     const dinero = createBigjsDinero;
     const bigjsUSD = castToBigjsCurrency(USD);
+    const bigjsMGA = castToBigjsCurrency(MGA);
+    const bigjsGBP = {
+      code: 'GBP',
+      base: [new Big(20), new Big(12)],
+      exponent: 1n,
```

Comment:
@sarahdayan I found a typing issue.
```suggestion
      exponent: new Big(1),
```